crates/iota-graphql-rpc/src/types/transaction_block/mod.rs

```diff
@@ -173,6 +219,7 @@ impl TransactionBlock {
     ///
     /// If the owner of the gas object(s) is not the same as the sender, the
     /// transaction block is a sponsored transaction block.
+    #[graphql(complexity = 0)]
     async fn gas_input(&self, ctx: &Context<'_>) -> Option<GasInput> {
```

Comment:
This contains sponsor address, and from the address user can query transactions blocks again.

```graphql
{
  transactionBlocks {
    edges {
      node {
        digest
        gasInput {
          gasSponsor {
            transactionBlocks {
              edges {
                node {
                  digest
                }
              }
            }
          }
        }
      }
    }
  }
}
```